telemetry/telemetry.go

```diff
@@ -219,113 +272,127 @@ func signalHandler(serverControlSignal chan<- int, wg *sync.WaitGroup, sigchanne
 	}
 }
 
-func startGNMIServer(telemetryCfg *TelemetryConfig, cfg *gnmi.Config, serverControlSignal <-chan int, wg *sync.WaitGroup) {
```

Comment:
><-chan int [](http://example.com/codeflow?start=90&length=10)

Do you really need to change type? I see iNotifyCertMonitoring requires old type.

Reply:
Yes, we will need to use a bidirectional channel here, since in StartGNMIServer goroutine we will be reading from serverControlSignal to stop/restart and we will also spawn monitoring goroutine which will be writing to serverControlSignal when cert files are deleted/modified.